### PR TITLE
Fix some Comment token issues

### PR DESCRIPTION
**Pull request by Thirumalai-Shaktivel**

### CODE REVIEW

File: src/lpython/parser/tokenizer.re [hunk 1]
```diff
@@ -448,13 +451,17 @@ int Tokenizer::lex(Allocator &al, YYSTYPE &yylval, Location &loc, diag::Diagnost
             }
 
             comment newline {
+                if(parenlevel) { continue; }
```
Comment (Thirumalai-Shaktivel) [inline src/lpython/parser/tokenizer.re:454]: `Comments` are completely ignored within the `parentheses`

File: src/lpython/parser/tokenizer.re [hunk 2]
```diff
@@ -448,13 +451,17 @@ int Tokenizer::lex(Allocator &al, YYSTYPE &yylval, Location &loc, diag::Diagnost
...
                 line_num++; cur_line=cur;
                 token(yylval.string);
                 yylval.string.n--;
                 token_loc(loc);
                 if (last_token == yytokentype::TK_NEWLINE) {
                     return yytokentype::TK_COMMENT;
                 } else {
+                    if (last_token == yytokentype::TK_COLON) {
+                        indent = true;
+                    }
```
Comment (Thirumalai-Shaktivel) [inline src/lpython/parser/tokenizer.re:464]: To handle the following  `Comments`
```python
for i in range(N): # type: parallel
```

**Review by certik — APPROVED** [review]
I think this looks good to merge. Thanks!

**Comment by Thirumalai-Shaktivel** [comment]
Thanks for reviewing and merging this PR!

**Pull request merged**